common/src/main/scala/explore/utils/package.scala

```diff
@@ -170,3 +176,81 @@ extension (bytes: NonNegLong)
 
 object IsExpanded extends NewType[Boolean]
 type IsExpanded = IsExpanded.Type
+
+extension [F[_], O](s: Stream[F, O])
+  /**
+   * Combine items within a given duration using Semigroup.
+   *
+   * When an item is received in the stream, a timer is started. Within that time, items are
+   * combined until the timer expires, at which point the combined item is emitted.
+   *
+   * If no items are received within the given duration, no item is emitted.
+   */
+  def reduceSemigroupWithin(d: FiniteDuration)(using
+    F: Temporal[F],
+    S: Semigroup[O]
+  ): Stream[F, O] =
+    reduceWithin(d, S.combine)
+
+  /**
+   * Combine items within a given duration using the given function.
+   *
+   * When an item is received in the stream, a timer is started. Within that time, items are
+   * combined until the timer expires, at which point the combined item is emitted.
+   *
+   * If no items are received within the given duration, no item is emitted.
+   *
+   * This might seem similar to `groupWithin`. The subtle difference is that in-between durations
+   * `groupWithin` enters a "timed out" state, where the next element will immediately be emitted,
+   * and only elements _after_ that will be grouped. This is not the case for `reduceWithin`, which
+   * will always wait for the timeout to expire before emitting the combined element.
```

Comment:
Thanks for adding the comment, sorry for being dumb, but I'm still struggling to see which of your tests specifically asserts this behavior.

Reply:
I've added a test that explicitly tests different behaviour. But another test already fails if I replace the implementation with groupWithin:

```
==> X explore.utils.StreamSpec.reduceWithin only combines within given time window 1.05s munit.ComparisonFailException: /Users/hvanrijswijk1/ws/explore/common/src/test/scala/explore/utils/StreamSpec.scala:32
31:
32:    TestControl.executeEmbed(program).assertEquals(Vector(4, 4, 4, 4, 4))
33:  }
values are not the same
=> Obtained
Vector(
  4,
  3,
  3,
  3,
  3
)
=> Diff (- obtained, + expected)
   4,
-  3,
-  3,
-  3,
-  3
+  4,
+  4,
+  4,
+  4
 )
```